src/ring_theory/ring_hom_properties.lean

```diff
@@ -0,0 +1,123 @@
+/-
+Copyright (c) 2022 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import algebra.category.Ring.constructions
+import category_theory.isomorphism
+import ring_theory.localization.away
+
+/-!
+# Properties of ring homomorphisms
+
+We provide the basic framework for talking about properties of ring homomorphisms.
+The following meta-properties are defined
+
+* `respects_iso`: `P` respects isomorphisms if `P f → P (e ≫ f)` and `P f → P (f ≫ e)`, where
+  `e` is an isomorphism.
+* `stable_under_composition`: `P` is stable under composition if `P f → P g → P (f ≫ g)`.
+* `stable_under_base_change`: `P` is stable under base change if `P (S ⟶ Y) → P (X ⟶ X ⊗[S] Y)`.
+
+-/
+
+universe u
+
+open category_theory opposite category_theory.limits
+
+namespace ring_hom
+
+variable (P : ∀ {R S : Type u} [comm_ring R] [comm_ring S] (f : by exactI R →+* S), Prop)
```

Comment:
Should this get a definition or notation, since it's quite long and appears many times?

Reply:
I think it only appeared twice; once per file?
I could still add such a definition if you think it is better to have it.

Reply:
You're right, then we can do without a definition.